### ISSUE TITLE
[Bug]: Handling ODT files not working as intended

### ISSUE THREAD

**Issue by roland4432** [pimcore/pimcore#13842]
### Expected behavior

Uploading an odt file should (corresponding to the manual: https://pimcore.com/docs/pimcore/current/Development_Documentation/Assets/Working_with_Thumbnails/Document_Thumbnails.html)
=> should lead to a working asset in the backend as well as a preview image

### Actual behavior

The file cannot be open in the backend nor can the preview be generated.

=> the type of the odt file is stored as "text" not as "document" (manually changing this in the DB at least let's one open the odt file in the backend but with an empty content)

Also the Pimcore\Model\Asset\Text does not have the method for generating a preview image via `getImageThumbnail`, which is correct. But the asset should be handled by Pimcore\Model\Asset\Document which is not being used do to the false detection of its type during upload.

### Steps to reproduce

* Create an *.odt file with libreoffice
* Upload it to the asset tree in pimcore
* Left click it to open => error
* Check database `assets`: type is text not document (caused by `pimcore/pimcore/models/Asset::getTypeFromMimeMapping`)

### Quick fix for correcting type detection
```php
#pimcore/pimcore/models/Asset.php
public static function getTypeFromMimeMapping($mimeType, $filename)
    {
        if ($mimeType == 'directory') {
            return 'folder';
        }

        $type = null;

        $mappings = [
            'document' => ['/msword/', '/pdf/', '/powerpoint/', '/office/', '/excel/', '/opendocument/'],
            'unknown' => ["/\.stp$/"],
            'image' => ['/image/', "/\.eps$/", "/\.ai$/", "/\.svgz$/", "/\.pcx$/", "/\.iff$/", "/\.pct$/", "/\.wmf$/", '/photoshop/'],
            'document' => ['/msword/', '/pdf/', '/powerpoint/', '/office/', '/excel/', '/opendocument/'], // move this above text since the mimetype for odt files is application/vnd.oasis.opendocument.text
            'text' => ['/text/', '/xml$/', '/\.json$/'],
            'audio' => ['/audio/'],
            'video' => ['/video/'],
            'archive' => ['/zip/', '/tar/'],
        ];

        foreach ($mappings as $assetType => $patterns) {
            foreach ($patterns as $pattern) {
                if (preg_match($pattern, $mimeType . ' .' . File::getFileExtension($filename))) {
                    $type = $assetType;

                    break;
                }
            }

            // break at first match
            if ($type) {
                break;
            }
        }

        if (!$type) {
            $type = 'unknown';
        }

        return $type;
    }
```

**Comment by brusch** [comment]
@roland4432 maybe we should change `'text' => ['/text/', '/xml$/', '/\.json$/'],` to `'text' => ['/text\//', '/xml$/', '/\.json$/'],`, that would be more precise, as only `text/` will be matched, wdyt? 

Can you create a PR for us? 😊

**Comment by roland4432** [comment]
Heya @brusch. I can easily fix the correct type for the asset. But the preview in backend is not working, nor the generation of a preview image for the document. (Both Ghostscript and Libreoffice are installed). 
Can you confirm?

I like the idea with the text/ regex since https://wiki.selfhtml.org/wiki/MIME-Type/%C3%9Cbersicht is pointing out, that all text based mimetypes are `@^text/@`

**Comment by dvesh3** [comment]
@roland4432 Hi, the preview is working on https://demo.pimcore.fun/admin. 

please make sure the maintenance is setup properly https://pimcore.com/docs/pimcore/current/Development_Documentation/Getting_Started/Installation.html#page_5-Maintenance-Cron-Job

**Comment by roland4432** [comment]
Heyho @dvesh3
We are currently not creating the thumbnails via cronjobs but on demand with `Asset\Service::getPreviewThumbnail` which is returning null.

I'll investigate further asap and provide more details on this topic

**Comment by brusch** [comment]
Fixed by #13952
